ksql-examples/src/main/java/io/confluent/ksql/datagen/ProducerFactory.java

```diff
@@ -15,26 +15,124 @@
 
 package io.confluent.ksql.datagen;
 
+import static java.util.Objects.requireNonNull;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.schema.ksql.PersistenceSchema;
 import io.confluent.ksql.serde.Format;
+import io.confluent.ksql.serde.KsqlSerdeFactories;
+import io.confluent.ksql.serde.KsqlSerdeFactory;
 import io.confluent.ksql.util.KsqlConfig;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.Properties;
+import java.util.function.Supplier;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.connect.data.ConnectSchema;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Struct;
+
+final class ProducerFactory {
+
+  private static final KsqlSerdeFactories SERDE_FACTORIES = new KsqlSerdeFactories();
+
+  private ProducerFactory() {
+  }
+
+  static DataGenProducer getProducer(
+      final Format keyFormat,
+      final Format valueFormat,
+      final Properties props
+  ) {
+    final KsqlSerdeFactory keyFactory = createSerdeFactory(keyFormat);
+    final KsqlSerdeFactory valueFactory = createSerdeFactory(valueFormat);
+
+    final Optional<SchemaRegistryClient> srClient = SchemaRegistryClientFactory
+        .getSrClient(keyFormat, valueFormat, props);
+
+    return new DataGenProducer(keyFactory, valueFactory, srClient::get);
+  }
+
+  private static KsqlSerdeFactory createSerdeFactory(final Format format) {
+    final KsqlSerdeFactory structSerdeFactory = SERDE_FACTORIES.create(format, Optional.empty());
+
+    if (format.supportsUnwrapping()) {
```

Comment:
did you mean `!format.supportsUnwrapping()`? are there unit tests for this?

Reply:
Nope, it's just the name of `UnwrappedSerdeFactory` that is wrong / misleading - should be `WrappedSerdeFactory` :D